server/routes/country_ambassador.js

```diff
@@ -0,0 +1,448 @@
+            let student = studentModel.toObject()
+            let userRecordWithAccessContext = await userSchema.findById(studentModel.user, {accessContexts: 1})
+            student.accessContexts = userRecordWithAccessContext.accessContexts
+            students.push(student)
+        }
+        res.status(200).send({'students': students})
+    } catch (e) {
+        console.log('ambassador/toggleModerator error: ' + e);
+        res.status(500).send({'ambassador/toggleModerator error' : e})
+    }
+});
+
+router.post('/addSchool/:alumniId/:country', passport.authenticate('jwt', {session: false}), async (req, res) => {
+    let alumniId = req.params.alumniId;
+    let country = req.body.country
+    let name = req.body.name
+    try {
+        if (!isCountryAmbassador(alumniId, country)) {
+            res.status(400).send('Alumnus does not have access as ambassador for ' + country);
+            return;
+        }
+        let school = new schoolSchema({
+            name: name,
+            country: country
+        })
+        await school.save();
+        res.status(200).send({'message': 'Successfully added school'})
+    } catch (e) {
+        console.log('/addSchool error:' + e);
+        res.status(500).send({'error' : 'Add School Error' + e})
+    }
+});
+
+router.get('/polls/:alumniId/:country', passport.authenticate('jwt', {session: false}), async (req, res) => {
+    let alumniId = req.params.alumniId;
+    let country = req.body.country
+    try {
+        if (!isCountryAmbassador(alumniId, country)) {
+            res.status(400).send('Alumnus does not have access as ambassador for ' + country);
+            return;
+        }
+        let polls = await pollSchema.find({countriesTargetted: {$in: [country]}}).populate('options schoolsTargetted')
+        res.status(200).json({
+            polls: polls
+        })
+    } catch (e) {
+        console.log('/polls error:' + e);
+        res.status(500).send({'error' : 'Fetching Polls' + e})
+    }
+});
+
+async function queuePolls(schoolsTargetted, countriesTargetted, rolesTargetted, pollModel) {
+    // by country
+    let usersToQueuePollsFor = []
+    let studentUsers = []
+    let alumniUsers = []
+    if (countriesTargetted.length) {
+        // BOTH alumni and students
+        if (rolesTargetted.includes('STUDENTS') && rolesTargetted.includes('ALUMNI')) {
+            studentUsers = await studentSchema.find().where('country').in(countriesTargetted).exec()
+            alumniUsers = await alumniSchema.find().where('country').in(countriesTargetted).exec()
+        } else if (rolesTargetted.includes('ALUMNI')) {
+            alumniUsers = await alumniSchema.find().where('country').in(countriesTargetted).exec()
+        } else if (rolesTargetted.includes('STUDENTS')) {
+            studentUsers = await studentSchema.find().where('country').in(countriesTargetted).exec()
+        }
+    }
+    // by school
+    else if (schoolsTargetted.length) {
```

Comment:
Looks like these two cases can be combined with just passing in either 'school' or 'country' into a function as the argument

Reply:
I think since we're just ever going to target by school or country, we'd be losing readability trading in maintainability like that. I don't think we're gaining too much abstracting away school vs country here, happy to have more back and forth on this!

Reply:
Hm, I'm not sure if I fully understand the context for what you're talking about here, but I guess it seems like those cases, even if they don't have 100% overlap, would have a lot of the same functionality? I guess if that ends up being the case, we could still use a function that does the part for the unified section where they're both the same, but then they could each have code that extends what they do beyond the shared part